src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,239 @@
+            throw new DataProcessingException("Couldn't update "
+                    + car + " in cars DB.", throwable);
+        }
+        removeDriversRelations(car);
+        List<Driver> drivers = car.getDrivers();
+        if (drivers != null) {
+            for (Driver driver : drivers) {
+                addDriverRelationForCar(car, driver);
+            }
+        }
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long carId) {
+        String query = "UPDATE cars SET deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, carId);
+            int numberOfDeletedRows = statement.executeUpdate();
+            return numberOfDeletedRows != 0;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't delete car with id " + carId, throwable);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getAllByDriverRequest = "SELECT * from cars_drivers cd "
+                + "LEFT JOIN cars c ON cd.car_id = c.id "
+                + "LEFT JOIN manufacturers m ON c.manufacturer_id = m.manufacturer_id "
+                + "WHERE driver_id = ? AND c.deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarsByDriverStatement
+                        = connection.prepareStatement(getAllByDriverRequest)) {
+            getCarsByDriverStatement.setLong(1, driverId);
+            ResultSet resultSet = getCarsByDriverStatement.executeQuery();
+            while (resultSet.next()) {
+                Car car = getCarWithManufacturerFromResultSet(resultSet);
+                cars.add(car);
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't get cars by drivers id " + driverId,
+                    throwable);
+        }
+        parseDriversToListOfCars(cars);
+        return cars;
+    }
+
+    private void removeDriversRelations(Car car) {
+        String deleteRequest = "DELETE FROM cars_drivers WHERE car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement removeStatement = connection.prepareStatement(deleteRequest)) {
+            removeStatement.setLong(1, car.getId());
+            removeStatement.executeUpdate();
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't delete drivers dependencies for car with id "
+                    + car.getId(), throwable);
+        }
+    }
+
+    private void addDriverRelationForCar(Car car, Driver driver) {
+        String insertDriverRelation
+                = "INSERT INTO cars_drivers (car_id, driver_id) VALUES (?, ?);";
```

Comment:
you have code duplication addDriverRelationForCar and insertDrivers are basically the same, try to think of the way to avoid having 2 methods

Reply:
done